src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt

```diff
@@ -21,52 +23,61 @@ import uk.gov.justice.digital.hmpps.visitscheduler.helper.visitNoteCreator
 import uk.gov.justice.digital.hmpps.visitscheduler.helper.visitSupportCreator
 import uk.gov.justice.digital.hmpps.visitscheduler.helper.visitVisitorCreator
 import uk.gov.justice.digital.hmpps.visitscheduler.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.visitscheduler.jpa.LegacyData
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.Visit
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitNoteType.VISITOR_CONCERN
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitNoteType.VISIT_COMMENT
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitNoteType.VISIT_OUTCOMES
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitRestriction
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitStatus
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.VisitType
+import uk.gov.justice.digital.hmpps.visitscheduler.jpa.repository.LegacyDataRepository
 import uk.gov.justice.digital.hmpps.visitscheduler.jpa.repository.VisitRepository
 import java.time.LocalDateTime
 
 class VisitResourceIntTest : IntegrationTestBase() {
   @Autowired
   private lateinit var visitRepository: VisitRepository
 
+  @Autowired
+  private lateinit var legacyDataRepository: LegacyDataRepository
+
   @AfterEach
   internal fun deleteAllVisits() = visitDeleter(visitRepository)
 
   @DisplayName("POST /visits")
   @Nested
   inner class CreateVisit {
-    private val createVisitRequest = CreateVisitRequest(
-      prisonId = "MDI",
-      prisonerId = "FF0000FF",
-      visitRoom = "A1",
-      visitType = VisitType.SOCIAL,
-      startTimestamp = visitTime,
-      endTimestamp = visitTime.plusHours(1),
-      visitStatus = VisitStatus.RESERVED,
-      visitRestriction = VisitRestriction.OPEN,
-      visitContact = CreateContactOnVisitRequest("John Smith", "01234 567890"),
-      visitors = listOf(CreateVisitorOnVisitRequest(123)),
-      visitorSupport = listOf(CreateSupportOnVisitRequest("OTHER", "Some Text")),
-      visitNotes = listOf(
-        VisitNoteDto(type = VISITOR_CONCERN, "My mother in-law is coming"),
-        VisitNoteDto(type = VISIT_OUTCOMES, "My mother wont visit again"),
-        VisitNoteDto(type = VISIT_COMMENT, "Mother in-law should be watched at all times")
+
+    fun createVisitRequest(leadPersonId: Long?): CreateVisitRequest {
```

Comment:
I think the original intention was to create a static structure for the test cases, you could achieve the same result by adding 'legacyData = CreateLegacyDataRequest(123)' to the end of the existing definition.

If you want to keep the change you can default the leadPersonId to null and save passing in null values in the call
fun createVisitRequest(leadPersonId: Long? = null): CreateVisitRequest { ....

I'm not a fan of this type of test templating unless its for mocking, might be worth having a backlog item to look into kotlin BDD frameworks.

Reply:
Will update